default/AI/FleetUtilsAI.py

```diff
@@ -320,41 +339,39 @@ def assess_fleet_role(fleet_id):
 
 
 def assess_ship_design_role(design):
-    if "CO_OUTPOST_POD" in design.parts:
+    parts = [fo.getPartType(partname) for partname in design.parts if partname]
+    if any((p.partClass == fo.shipPartClass.colony and p.capacity == 0) for p in parts):
         if design.starlaneSpeed > 0:
             return AIShipRoleType.SHIP_ROLE_CIVILIAN_OUTPOST
         else:
             return AIShipRoleType.SHIP_ROLE_BASE_OUTPOST
 
-    if "CO_COLONY_POD" in design.parts or "CO_SUSPEND_ANIM_POD" in design.parts:
+    if any((p.partClass == fo.shipPartClass.colony and p.capacity > 0 for p in parts)):
         if design.starlaneSpeed > 0:
             return AIShipRoleType.SHIP_ROLE_CIVILIAN_COLONISATION
         else:
             return AIShipRoleType.SHIP_ROLE_BASE_COLONISATION
 
-    if "GT_TROOP_POD" in design.parts:
+    if any((p.partClass == fo.shipPartClass.troops for p in parts)):
         if design.starlaneSpeed > 0:
             return AIShipRoleType.SHIP_ROLE_MILITARY_INVASION
         else:
             return AIShipRoleType.SHIP_ROLE_BASE_INVASION
 
     if design.starlaneSpeed == 0:
-        if len(design.parts) == 0 or design.parts[0] in ["SH_DEFENSE_GRID", "SH_DEFLECTOR", "SH_MULTISPEC"] or (len(design.parts) == 1 and design.parts[0] == ''):
+        if len(parts) == 0 or parts[0].partClass == fo.shipPartClass.shields:
```

Comment:
Unless I'm overlooking something, Cj, design.parts has not been wrapped with anything to make it return a python list, and so it is still returning a C++ container wrapped with the boost indexing suite.  To my knowledge this does not automatically use container.empty() as a boolean conversion.  So the length check is still necessary.